packages/editor-ui/src/stores/assistant.store.ts

```diff
@@ -601,20 +601,18 @@ export const useAssistantStore = defineStore(STORES.ASSISTANT, () => {
 				(msg) => onEachStreamingMessage(msg, id),
 				() => onDoneStreaming(id),
 				(e) =>
-					handleServiceError(
-						e,
-						id,
-						async () => await withoutFailedMessages(async () => await sendMessage(chatMessage)),
-					),
+					handleServiceError(e, id, async () => {
+						chatMessages.value = chatMessages.value.filter((msg) => msg.id !== id);
+						await sendMessage(chatMessage);
+					}),
 			);
 			trackUserMessage(chatMessage.text, !!chatMessage.quickReplyType);
 		} catch (e: unknown) {
 			// in case of assert
-			handleServiceError(
-				e,
-				id,
-				async () => await withoutFailedMessages(async () => await sendMessage(chatMessage)),
-			);
+			handleServiceError(e, id, async () => {
+				chatMessages.value = chatMessages.value.filter((msg) => msg.id !== id);
```

Comment:
so pretty, much simpler

Any reason we cannot just do this for every service error in `handleServiceError`?

Reply:
I mostly liked this being more explicit since it's closer to the source where it matters. Semantically this way each function that calls handleServiceError defines which operations are necessary to undo itself.

On the other hand moving this down a level would future-proof us more. Let me know what you prefer, I'm indifferent.

Reply:
it's your call, not mine. I would simplify to reduce duplication here.

Reply:
I tried to implement yours but there's actually an awkward case since we don't want to display the button for the theoretical case where we don't have a retry button, and maintaining this property becomes pretty awkward, we'd have to do `retry && async () => { /*logic*/;  await retry(); }` :/

I agree on the duplication being bad, I'll move them to a local `retry` function at least.

Reply:
it's fine if it's too much work. don't want to make things complicated

Reply:
I'm happy with the local function, should be ready to go now